test/exitHandler.js

```diff
@@ -258,6 +308,114 @@ contract('ExitHandler', (accounts) => {
         
       });
 
+      it('Should allow to challenge exit by verified tx', async () => {
+        await time.advanceBlock();
+        transferTx = Tx.transfer(
+          [new Input(new Outpoint(depositTx.hash(), 0))],
+          [new Output(50, alice), new Output(50, alice)]
+        ).sign([alicePriv]);
+        const spendTx = Tx.transfer(
+          [new Input(new Outpoint(transferTx.hash(), 1))],
+          [new Output(25, alice), new Output(25, alice)]
+        ).sign([alicePriv]);
+        const consolidateTx = Tx.consolidate([
+          new Input(new Outpoint(transferTx.hash(), 0)),
+          new Input(new Outpoint(spendTx.hash(), 0)),
+          new Input(new Outpoint(spendTx.hash(), 1))
+        ], new Output(100, alice));
+        const period = await submitNewPeriod([depositTx, transferTx, spendTx, consolidateTx]);
+
+        const transferProof = period.proof(transferTx);
+        const spendProof = period.proof(spendTx);
+        const consolidateProof = period.proof(consolidateTx);
+
+        // withdraw output
+        await exitHandler.startExit(transferProof, spendProof, 1, 0, { from: alice });
+
+        await exitHandler.startVerification(transferProof, 0);
```

Comment:
so there is one verfication per input and verifications are submitted by challenger, right? Is it possible, that I as attacker will fragment my UTXO set and consolidate them including invalid inputs. Challenge will require submitting a lot of txs thus making it too costly for challenger

Reply:
> consolidate them including invalid inputs
we assume the moment that the operator includes invalid transactions, everyone will start exiting the chain. so in the scenario you describe above, there is no need to challenge, you simply exit because the chain became invalid.

the verification is only there to deny the operator the possibility to:
a) challenge valid exits.
b) decrease the priority of funds in exit queue by consolidating them,
       putting the priority lower than his own transaction in the latest invalid block.

i'm not sure if the limit of 16 inputs for consolidate is actually implemented somewhere. but i guess that would be the max transaction that need to be submitted to challenge with the biggest consolidate tx possible.